pkgs/by-name/bt/btrfs-list/package.nix

```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  btrfs-progs,
+  perl,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "btrfs-list";
+  version = "2.3";
+
+  src = fetchFromGitHub {
+    owner = "speed47";
+    repo = "btrfs-list";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-cWzDRop0cyrjVIJzuZxTqELgec66GiPAUJY1xIBr3OY=";
+  };
+
+  buildInputs = [ perl ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    install -D -t $out/bin btrfs-list
+
+    wrapProgram $out/bin/btrfs-list --prefix PATH : ${lib.makeBinPath [ btrfs-progs ]}
+  '';
```

Comment:
```suggestion

  # check "run_cmd" invocation
  runtimeDeps = [
    coreutils
    ncurses # tput
    btrfs-progs
  ];

  installPhase = ''
    install -D -t $out/bin btrfs-list

    wrapProgram $out/bin/btrfs-list --prefix PATH : ${lib.makeBinPath finalAttrs.runtimeDeps}
  '';
```

Reply:
What's this `runtimeDeps` attribute? Does it have any special behaviour in the context of `stdenv.mkDerivation`?

You're spot on with the missing `ncurses` deps, thanks.

What's `coreutils` useful for? I can run the command without that.

Reply:
> What's this `runtimeDeps` attribute? Does it have any special behaviour in the context of `stdenv.mkDerivation`? If not, we should move it to a `let` binding IMO.

No, because we are making use of `finalAttrs` attribute, so it is overridable.

> What's `coreutils` useful for? I can run the command without that.

To prevent behavioral differences caused by impurity to the greater extent, such as on other distributions.

Reply:
> To prevent behavioral differences caused by impurity to the greater extent, such as on other distributions.

Is it because of [this](https://github.com/speed47/btrfs-list/blob/30bb31e0a7e3aef08bc30a81fedb34d34be9f1e4/btrfs-list#L211C43-L211C45) usage of `readlink`?

Reply:
I mean "generally this is a good practice" (considering that we have distributions using busybox and even freebsd utils)

Reply:
I'm quite fine to use it in this case (cause of `readlink`), but how general do you mean when you say "generally"? Every Linux package?

Reply:
Also what does the `check "run_cmd" invocation` comment mean?

Reply:
I chose to inline the runtime deps, instead of defining an attribute for them, because IIUC that attribute would end up in the derivation, which seems unclean to me. The other reason is that I don't see a need for allowing overriding the set of runtime dependencies for this package specifically -- if it's something we want to do for all (or a subset) of packages, we should do it more formally, and document it, IMO.